python/ray/serve/driver.py

```diff
@@ -0,0 +1,76 @@
+import inspect
+from abc import abstractmethod
+from typing import Any, Callable, Optional, Union
+
+import starlette
+from fastapi import Depends, FastAPI
+
+from ray._private.utils import import_attr
+from ray.ml.predictor import DataBatchType
+from ray.serve.api import RayServeDAGHandle
+from ray.serve.http_util import ASGIHTTPSender
+from ray import serve
+
+DEFAULT_INPUT_SCHEMA = "ray.serve.http_adapters.array_to_databatch"
+InputSchemaFn = Callable[[Any], DataBatchType]
+
+
+def _load_input_schema(
+    input_schema: Optional[Union[str, InputSchemaFn]]
+) -> InputSchemaFn:
+    if input_schema is None:
+        input_schema = DEFAULT_INPUT_SCHEMA
+
+    if isinstance(input_schema, str):
+        input_schema = import_attr(input_schema)
+
+    assert inspect.isfunction(input_schema), "input schema must be a callable function."
+    return input_schema
+
+
+class SimpleSchemaIngress:
+    def __init__(self, input_schema: Optional[Union[str, InputSchemaFn]] = None):
+        """Create a FastAPI endpoint annotated with input_schema dependency.
+
+        Args:
+            input_schema(str, InputSchemaFn, None): The FastAPI input conversion
+              function. By default, Serve will use the `NdArray` schema and convert to
+              numpy array. You can pass in any FastAPI dependency resolver that returns
+              an array. When you pass in a string, Serve will import it.
+              Please refer to Serve HTTP adatper documentation to learn more.
+        """
+        input_schema = _load_input_schema(input_schema)
+        self.app = FastAPI()
+
+        @self.app.get("/")
+        @self.app.post("/")
+        async def handle_request(inp=Depends(input_schema)):
```

Comment:
In current serve i think we're default to http endpoint without requiring FastAPI as dependency but our default driver seems to always require it. Is there a minimal http equivalent of pipeline dag ?

Reply:
There is a tradeoff here. Should we default to let user write the http handling code (existing state) or just provide a default. We can potentially just set the default schema to `starlette request -> request` to match existing state. Thoughts? @jiaodong @edoakes

Reply:
FastAPI is a required dependency, so I don't really see the reason not to use it by default.

Reply:
But should the default input be starlette or numpy right now?

Reply:
I do have a preference for staying with starlette but since this is a brand new API 🤷‍♂️

Reply:
resolved offline; starlette by default